### PR TITLE
periodic: account for differences with Konflux built images

### PR DESCRIPTION
**Pull request by crozzy**
For the RPM test we check the index report against the pyxis manifest for that image. The format of this manifest has recently changed when they started producing builds from Konflux, this change accounts for those changes.

### CODE REVIEW

**Comment by crozzy** [comment]
closes https://github.com/quay/claircore/issues/1444

File: test/rpmtest/manifest.go [hunk 1]
```diff
@@ -44,15 +45,28 @@ func PackagesFromRPMManifest(t *testing.T, r io.Reader) []*claircore.Package {
...
-		if s, ok := src[rpm.Source]; ok {
+
+		// Newer images produced from Konflux shove all the source information
+		// into the SourceName and omit the SourceNEVRA. Try both.
+		var source string
+		switch {
+		case rpm.SourceNEVRA != "":
+			source = rpm.SourceNEVRA
+		case rpm.SourceName != "":
+			source = rpm.SourceName
```
Comment (RTann) [inline test/rpmtest/manifest.go:56]: is it possible for both to be populated?

File: test/rpmtest/manifest.go [hunk 2]
```diff
@@ -44,15 +45,28 @@ func PackagesFromRPMManifest(t *testing.T, r io.Reader) []*claircore.Package {
...
+		default:
+			continue
+		}
+
+		if s, ok := src[source]; ok {
 			p.Source = s
 		} else {
-			s := strings.TrimSuffix(rpm.Source, ".src")
+			s := strings.TrimSuffix(strings.TrimSuffix(source, ".rpm"), ".src")
```
Comment (RTann) [inline test/rpmtest/manifest.go:64]: so it's either `.src` or `.src.rpm`?

File: test/rpmtest/manifest.go [hunk 3]
```diff
@@ -44,15 +45,28 @@ func PackagesFromRPMManifest(t *testing.T, r io.Reader) []*claircore.Package {
...
 			pos := len(s)
 			for i := 0; i < 2; i++ {
 				pos = strings.LastIndexByte(s[:pos], '-')
 				if pos == -1 {
-					t.Fatalf("malformed NEVRA: %q", rpm.Source)
+					t.Fatalf("malformed NEVRA: %q for %q", source, rpm.Name)
```
Comment (RTann) [inline test/rpmtest/manifest.go:69]: nit: may not be NEVRA

File: test/rpmtest/manifest.go (hunk 1 above)
Comment (crozzy) [reply test/rpmtest/manifest.go:56]: Yes, older manifests will have `SourceNEVRA` populated, newer ones won't but they have the same information in `SourceName`.

File: test/rpmtest/manifest.go (hunk 2 above)
Comment (crozzy) [reply test/rpmtest/manifest.go:64]: yeah, from the data the `SourceNEVRA` will have the `.rpm` , `SourceName` will have `.src.rpm`

File: test/rpmtest/manifest.go (hunk 3 above)
Comment (crozzy) [reply test/rpmtest/manifest.go:69]: We're expecting something like either 
```
acl-2.3.1-4.el9.src.rpm // SourceName
acl-0:2.3.1-3.el9.src      // SourceNEVRA
```
So I guess yeah, one is NEVRA and one is NVRA

**Review by RTann — APPROVED** [review]

**Pull request merged**